cdap-watchdog/src/main/java/io/cdap/cdap/logging/serialize/DelegatingLoggingEvent.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.logging.serialize;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.classic.spi.IThrowableProxy;
+import ch.qos.logback.classic.spi.LoggerContextVO;
+import org.slf4j.Marker;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class DelegatingLoggingEvent implements ILoggingEvent {
+
+  private ILoggingEvent loggingEvent;
+  private Map<String, String> mdc = new HashMap<>();
+
+  public DelegatingLoggingEvent(ILoggingEvent event, Map<String, String> mdc) {
+    this.loggingEvent = event;
+    this.mdc.putAll(mdc);
```

Comment:
should wrap this in an unmodifiable collection. In almost all cases, getter methods should be returning unmodifiable collections.

`this.mdc = Collections.unmodifiableMap(new HashMap<>(mdc))`

Reply:
Sure, I will take care of this.